magicctapipe/scripts/lst1_magic/semi_automatic_scripts/coincident_events.py

```diff
@@ -0,0 +1,374 @@
+"""
+This scripts facilitates the usage of the script
+"lst1_magic_event_coincidence.py". This script is
+more like a "manager" that organizes the analysis
+process by:
+1) Creating the bash scripts for looking for
+coincidence events between MAGIC and LST in each
+night.
+2) Creating the subdirectories for the coincident
+event files.
+
+Usage:
+$ python coincident_events.py (-c config_file.yaml)
+"""
+import argparse
+import glob
+import logging
+import os
+from datetime import date as dtdt
+from datetime import timedelta
+from pathlib import Path
+
+import numpy as np
+import yaml
+
+from magicctapipe import __version__
+
+__all__ = ["configfile_coincidence", "linking_bash_lst"]
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def configfile_coincidence(ids, target_dir):
+
+    """
+    This function creates the configuration file needed for the event coincidence step
+
+    Parameters
+    ----------
+    ids : list
+        List of telescope IDs
+    target_dir : str
+        Path to the working directory
+    """
+
+    lines = [
+        f"mc_tel_ids:\n    LST-1: {ids[0]}\n    LST-2: {ids[1]}\n    LST-3: {ids[2]}\n    LST-4: {ids[3]}\n    MAGIC-I: {ids[4]}\n    MAGIC-II: {ids[5]}\n\n",
+        'event_coincidence:\n    timestamp_type_lst: "dragon_time"  # select "dragon_time", "tib_time" or "ucts_time"\n    pre_offset_search: true\n    n_pre_offset_search_events: 100\n    window_half_width: "300 ns"\n',
+        '    time_offset:\n        start: "-10 us"\n        stop: "0 us"\n',
+    ]
+    with open(f"{target_dir}/config_coincidence.yaml", "w") as f:
+        f.writelines(lines)
+
+
+def linking_bash_lst(
+    target_dir, LST_runs, nsb, date, source, LST_version, env_name, NSB_match
+):
+
+    """
+    This function links the LST data paths to the working directory and creates bash scripts.
+
+    Parameters
+    ----------
+    target_dir : str
+        Path to the working directory
+    LST_runs : matrix of strings
+        This matrix is imported from config_general.yaml and tells the function where to find the LST data and link them to our working directory
+    nsb : int
+        NSB level
+    date : numpy array
+        Array of lists [date run] for all the LST runs (before the NSB splitting)
+    source : str
+        Target name
+    LST_version : str
+        The lstchain version used to process the LST data
+    env_name : str
+        Name of the conda environment
+    NSB_match : bool
+        If real data are matched to pre-processed MCs or not
+    """
+
+    ST_list = [
+        os.path.basename(x) for x in glob.glob(f"{target_dir}/v{__version__}/DL1/*")
+    ]
+
+    if (len(LST_runs) == 2) and (len(LST_runs[0]) == 10):
```

Comment:
this looks strange

Reply:
Do you mean line 88? In case there is only one entry (i.e., one line) in the MAGIC_runs.txt or LST_runs.txt, LST_runs won't be a set of [date,run] couples (i.e., list of lists  [[xx,yy],[xx,yy]....]) but will be read as a list of two elements ([date,run]) and will thus split run and date instead of using them as a couple; this if clause fixes this issue by 'adding two brackets' around the list in the latter case. This way, we will always deal with [[...]....] objects. But any suggestion here is welcome